pagopa/io-app
# chore: [IOPID-2329] Improve `Keychain` errors also leveraging Sentry

## Short description
This pull request includes several changes (as described [here](https://pagopa.atlassian.net/wiki/spaces/IAEI/pages/1291092330/Analisi+per+miglioramento+tracciamento+errore+keychain)) to improve error tracking and handling for keychain operations, as well as some updates to the Sentry integration. 

Error tracking improvements:

* [`ts/store/storages/keychain.ts`](diffhunk://#diff-736cc87245dcde2f05bdff5cee1a0454114050d8f39a7b491929e384bf2360fbL13-R35): Introduced specific error tracking for keychain operations using Sentry and updated error variables to `getKeychainError`, `setKeychainError`, and `removeKeychainError`. [[1]](diffhunk://#diff-736cc87245dcde2f05bdff5cee1a0454114050d8f39a7b491929e384bf2360fbL13-R35) [[2]](diffhunk://#diff-736cc87245dcde2f05bdff5cee1a0454114050d8f39a7b491929e384bf2360fbR48-R90)
* [`ts/utils/analytics.ts`](diffhunk://#diff-a425aab9658df7526ca2afd44dfd12ab19227f77c49e80436ef35388f2266627L184-R206): Updated `trackKeychainFailures` to track different keychain errors and clear them after tracking.

Sentry integration updates:

* [`ts/App.tsx`](diffhunk://#diff-10dd2581dc178023f71c4757443c768e5e1f61b3f83998c2ffacdb99740a375dL47-R51): Replaced the deprecated `new Sentry.ReactNativeTracing` with `Sentry.reactNativeTracingIntegration` and added `maxValueLength` to the Sentry configuration so that message values are truncated after `3000` characters instead of `250` (the default value).

`initializeApplicationSaga.test.ts` updates:

* [`ts/sagas/__tests__/initializeApplicationSaga.test.ts`](diffhunk://#diff-8ae536898e67d19b3d0746135b63e747cc3b2af485fa7e340722dd4ff2447badR48): update tests with`trackKeychainFailures`.  [[1]](diffhunk://#diff-8ae536898e67d19b3d0746135b63e747cc3b2af485fa7e340722dd4ff2447badR48) [[2]](diffhunk://#diff-8ae536898e67d19b3d0746135b63e747cc3b2af485fa7e340722dd4ff2447badR120) [[3]](diffhunk://#diff-8ae536898e67d19b3d0746135b63e747cc3b2af485fa7e340722dd4ff2447badR175) [[4]](diffhunk://#diff-8ae536898e67d19b3d0746135b63e747cc3b2af485fa7e340722dd4ff2447badR223) [[5]](diffhunk://#diff-8ae536898e67d19b3d0746135b63e747cc3b2af485fa7e340722dd4ff2447badR276) [[6]](diffhunk://#diff-8ae536898e67d19b3d0746135b63e747cc3b2af485fa7e340722dd4ff2447badR341) [[7]](diffhunk://#diff-8ae536898e67d19b3d0746135b63e747cc3b2af485fa7e340722dd4ff2447badR395)

Code cleanup:

* [`ts/sagas/startup.ts`](diffhunk://#diff-8a5b2f3967d681b976fe673762bd1061f5b430130c880c1195b76af06362cf31L100-R100): Removed old keychain error tracking code and replaced it with the new `trackKeychainFailures` function. [[1]](diffhunk://#diff-8a5b2f3967d681b976fe673762bd1061f5b430130c880c1195b76af06362cf31L100-R100) [[2]](diffhunk://#diff-8a5b2f3967d681b976fe673762bd1061f5b430130c880c1195b76af06362cf31L118-L121) [[3]](diffhunk://#diff-8a5b2f3967d681b976fe673762bd1061f5b430130c880c1195b76af06362cf31R256-R259) [[4]](diffhunk://#diff-8a5b2f3967d681b976fe673762bd1061f5b430130c880c1195b76af06362cf31L503-L507)


## How to test
- Run the application and  test not to have a regression in the session persisted state.
- Enable sentry in debug mode and set the `sampleRate` to `1`. Then try to mock a native exception in the react-native-keychain module and see if both mixpanel events are tracked and the exception is sent to Sentry (ex. [message](https://pagopaspa.sentry.io/issues/9134740/events/?alert_rule_id=9449&alert_type=issue&notification_uuid=b410db73-df7d-4442-b310-2ddb591fff16&project=4507221483585616&referrer=slack), [exception](https://pagopaspa.sentry.io/issues/9129567/events/04c33b0b3c6049c99b252bcafc6770e8/?project=4507221483585616&referrer=issue_details.related_trace_issue))
